academy/xl-run-prod/3-genesis.md

```diff
@@ -7,8 +7,203 @@ tag: deep-dive
 
 # Prepare where the node starts
 
+When you run your chain with `myprojectd start`, the software will build from a genesis. This genesis needs to contain:
+
+1. The original validators and their stakes.
+2. The original state values that you see fit. This can be, for instance, the original parameters of your world if you are building a metaverse, or the original USD price of your token if you are building a stable coin.
+
 ## Generate
 
+You are going to build this genesis file progressively. The genesis contains a lot of information such as [minting](https://hub.cosmos.network/main/resources/genesis.html#mint), [staking](https://hub.cosmos.network/main/resources/genesis.html#staking) or [slashing](https://hub.cosmos.network/main/resources/genesis.html#slashing) parameters. You can adjust [these parameters](https://hub.cosmos.network/main/resources/genesis.html) if you know what you are doing.
+
+<!--
+
+TODO add link that lists what can be changed in genesis.
+
+-->
+
+The genesis needs only be generated once, then it is distributed and copied onto machines.
+
+### Token name(s)
+
+One of the simpler parameters that you can decide on is the name of the staking token. The name you choose is for the indivisible denomination. For the Cosmos Hub that is `uatom`, which is understood as a millionth of an ATOM.
+
+For instance, if you decide that your token is the STONE and that it contains a billion indivisible units, by convention, you would name the unit an `nstone`, as in a _nano STONE_.
+
+Keeping on-brand, your new chain is named _stone-age_.
+
+To create a brand new genesis file for your chain, run:
+
+```sh
+$ ./myprojectd init --chain-id stone-age --staking-bond-denom nstone
+```
+
+This creates a large genesis file with default values mirrored from the Cosmos Hub.
+
+<HighlightBox type="remember">
+
+Correctly identify where it is located. Most likely in `~/.myprojectd/config/genesis.json`. If this is not where you want to prepare your genesis file, you can add `--home another_folder` to all your commands.
+
+</HighlightBox>
+
+If you are planning on having more denominations than your staking token, this is defined in the next step.
+
+### Genesis accounts
+
+Genesis accounts are accounts that exist in the genesis. They can be there because of a pre-sale of tokens, because of simple allotments or for any other reason. You need to include them in the genesis. The command for that is `add-genesis-account`. For each genesis account you need to collect their address. If these are third-parties, make sure in more ways than one that you get the right values, whether by email or other means.
+
+Here you can decide to allocate new tokens to your genesis accounts. There is no limit to the number of genesis accounts and the number of extra tokens in the genesis. If you introduce another token name `nflint`, then if Alice has the address `cosmos1nw793j9xvdzl2uc9ly8fas5tcfwfetercpdfqq` you could make her a genesis account with:
```

Comment:
Isn't it wise for personalization’s sake to change, as well, the address prefix? Maybe this is not the right tutorials to explain how (https://github.com/osmosis-labs/osmosis/blob/main/app/params/config.go#L10-L32) but it feels weird to is `cosmos...` addresses in a tutorial for a own chain.

Reply:
Yes, I was thinking about that for Checkers too. Since I use chain id "stone age" and coin "flint", maybe we could use the prefix of `caveman` or something like that :)